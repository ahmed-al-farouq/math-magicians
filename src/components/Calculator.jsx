import React from 'react';
import PropTypes from 'prop-types';
import calculate from '../logic/calculate';

function Calculator({ data, value, updateState }) {
  const updateData = (e) => {
    const newData = calculate(data, e.target.textContent);
    updateState(newData);
  };

  return (
    <section className="calculator-container">
      <div className="output">
        <div className="operand">
          {value || 0}
        </div>
      </div>
      <button type="button" onClick={updateData}>AC</button>
      <button type="button" onClick={updateData}>+/-</button>
      <button type="button" onClick={updateData}>%</button>
      <button type="button" onClick={updateData} className="operators">รท</button>
      <button type="button" onClick={updateData}>7</button>
      <button type="button" onClick={updateData}>8</button>
      <button type="button" onClick={updateData}>9</button>
      <button type="button" onClick={updateData} className="operators">x</button>
      <button type="button" onClick={updateData}>4</button>
      <button type="button" onClick={updateData}>5</button>
      <button type="button" onClick={updateData}>6</button>
      <button type="button" onClick={updateData} className="operators">+</button>
      <button type="button" onClick={updateData}>1</button>
      <button type="button" onClick={updateData}>2</button>
      <button type="button" onClick={updateData}>3</button>
      <button type="button" onClick={updateData} className="operators">-</button>
      <button type="button" onClick={updateData} className="span-two">0</button>
      <button type="button" onClick={updateData}>.</button>
      <button type="button" onClick={updateData} className="operators">=</button>
    </section>
  );
}

Calculator.propTypes = {
  data: PropTypes.objectOf(PropTypes.object).isRequired,
  value: PropTypes.string.isRequired,
  updateState: PropTypes.func.isRequired,
};

export default Calculator;
