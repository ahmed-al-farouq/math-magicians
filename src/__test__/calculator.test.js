import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('Match calculator snapshoot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
