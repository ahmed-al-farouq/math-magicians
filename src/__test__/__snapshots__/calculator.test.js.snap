// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match calculator snapshoot 1`] = `
<section
  className="sc-bdvvtL gEHaGq"
>
  <h2
    className="sc-gsDKAQ dCIjQH"
  >
    Let’s do some maths!
  </h2>
  <div
    className="calculator-container"
  >
    <div
      className="output"
    >
      <div
        className="operand"
      >
        0
      </div>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operators"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operators"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operators"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operators"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="span-two"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operators"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;
